src/core/thread/link_metrics.cpp

```diff
@@ -164,83 +299,205 @@ otError LinkMetrics::AppendLinkMetricsReport(Message &aMessage, const Message &a
     return error;
 }
 
+otError LinkMetrics::HandleLinkMetricsManagementRequest(const Message &      aMessage,
+                                                        Neighbor &           aNeighbor,
+                                                        otLinkMetricsStatus &aStatus)
+{
+    otError       error = OT_ERROR_NONE;
+    Tlv           tlv;
+    uint8_t       seriesId;
+    SeriesFlags   seriesFlags;
+    otLinkMetrics linkMetrics;
+    bool          doForwardProbingRegistration = false;
+    uint16_t      offset;
+    uint16_t      length;
+    uint16_t      index = 0;
+
+    SuccessOrExit(error = Tlv::FindTlvValueOffset(aMessage, Mle::Tlv::Type::kLinkMetricsManagement, offset, length));
+
+    while (index < length)
+    {
+        uint16_t pos = offset + index;
+        SuccessOrExit(aMessage.Read(pos, tlv));
+
+        pos += sizeof(tlv);
+
+        switch (tlv.GetType())
+        {
+        case kForwardProbingRegistration:
+            VerifyOrExit(!doForwardProbingRegistration, error = OT_ERROR_PARSE);
+            SuccessOrExit(aMessage.Read(pos, seriesId));
+            pos += sizeof(seriesId);
+            SuccessOrExit(aMessage.Read(pos, seriesFlags));
+            pos += sizeof(seriesFlags);
+            SuccessOrExit(error = ReadTypeIdFlagsFromMessage(
+                              aMessage, pos, static_cast<uint16_t>(offset + index + tlv.GetSize()), linkMetrics));
+            doForwardProbingRegistration = true;
+            break;
+
+        default:
+            break;
+        }
+
+        index += tlv.GetSize();
+    }
+
+    if (doForwardProbingRegistration)
+    {
+        aStatus = ConfigureForwardTrackingSeries(seriesId, seriesFlags, linkMetrics, aNeighbor);
```

Comment:
`aStatus` is not being set when the request is malformed.

Reply:
If the request is malformed, this function would exit earlier (in the before `while loop`). So if the code reaches here, the message must be well-formed. And if the request is malformed, nothing would be done further and no response would be sent so `aStatus` is not required in this case.

Reply:
Upd: I think this comment is actually caused by the next comment. I have updated so that when the request is malformed it would be simply discarded. So the `aStatus` is not required.